src/analyser.cpp

```diff
@@ -272,7 +274,7 @@ bool AnalyserInternalEquation::check(size_t &equationOrder, size_t &stateIndex,
         for (size_t i = 0; i < mComponent->variableCount(); ++i) {
             auto localVariable = mComponent->variable(i);
 
-            if (isSameOrEquivalentVariable(variable->mVariable, localVariable)) {
+            if (model->areEquivalentVariables(variable->mVariable, localVariable)) {
```

Comment:
(ditto below)

Reply:
FWIW, it's not part of the `Model` class, but the `AnalyserModel` class. So, yes, we now have two `areEquivalentVariables()` methods. One in our utilities and another in `AnalyserModel`. The difference between the two versions is that the latter relies on caching (something that can be done since an `AnalyserModel` is a static model unlike a `Model` which is "live"). No, the reason that the second `areEquivalentVariables()` method is in `AnalyserModel`, well... that's the only place I can think of, but happy to consider another location.